src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/StageDetail/StageDetail.component.js

```diff
@@ -65,6 +82,18 @@ const StageDetailPlain = ({ events, eventName, dataElements, classes }: Props) =
         });
     };
 
+    const handleViewAll = () => {
+        history.push(
+            `/enrollment/stageEvents?${urlArguments({ orgUnitId, programId, stageId })}`,
```

Comment:
Hey @huocha, 

I think both navigations handlers should be moved at the Enrollment page level.  
The `WidgetsStagesAndEvents` can expose callbacks when the user clicks the `New {{ eventName }} event` and `Go to full {{ eventName }}` buttons. Using this strategy `WidgetsStagesAndEvents` is decoupled from the `Enrollment` page routing logic. The widget can be easier reused in other pages that will handle their own navigation. 

What are your thoughts on this? Thank you!

Reply:
hey @simonadomnisoru, I don't think it's possible to handle the navigation from the enrollment page since it needs the `stageId` which only specified in the `WidgetStagesAndEvents`

Reply:
The `stageId` can be passed as a parameter to the callback function exposed from `WidgetStagesAndEvents` and used on the page. 


1) In the `Enrollment` page:

`const handleOnClick = (stageid) => history.push(/enrollment/stageEvents?${urlArguments({ orgUnitId, programId, stageId })}`
....
`<WidgetStagesAndEvents callbackOnClick={callbackOnClick} .... /> `


2) In the WidgetStagesAndEvents/Stages/Stage/StageDetail/StageDetail.component.js the callback will be used like this.
```
  const handleViewAll = () => {
        const {callbackOnClick} = props;
        callbackOnClick(stageId)
}
```

What do you think?

Reply:
I can not imagine these buttons to behave differently as their labels said "Go to this page" or "Create new event", it also attached to the widget with the `stageId` so I don't think pass the callback 3 levels above worth it. 
But I will let the team decides if we should do it or not
@JoakimSM @EirikHaugstulen WDYT?

Reply:
hey @simonadomnisoru, after discussing with @JoakimSM we agree to handle it with the callback.
I will change the implementation.